code/game/mecha/mecha_defense.dm

```diff
@@ -1,7 +1,14 @@
+/obj/mecha/proc/get_armour_facing(srcdir,adir)
+	var/facing_hit = armour_facings["[srcdir]"]["[adir]"]
+	var/damage_modifier = facing_modifiers[facing_hit]
+	return damage_modifier
 
-/obj/mecha/proc/take_damage(amount, type="brute")
+/obj/mecha/proc/take_damage(amount, type="brute", adir = 0)
 	if(amount)
 		var/damage = absorbDamage(amount,type)
+		if(adir)
+			var/modifier = get_armour_facing(src.dir,adir)
+			damage = damage/modifier
```

Comment:
use multiplication, easier to manipulate, less chance of a divide by zero error.